app/scripts/lib/transaction/util.ts

```diff
@@ -67,17 +68,23 @@ export type AddTransactionRequest = FinalAddTransactionRequest & {
 };
 
 export type AddDappTransactionRequest = BaseAddTransactionRequest & {
-  // TODO: Fix in https://github.com/MetaMask/metamask-extension/issues/31973
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  dappRequest: Record<string, any>;
+  dappRequest: JsonRpcRequest;
+  requestContext: MiddlewareContext;
 };
 
 export async function addDappTransaction(
   request: AddDappTransactionRequest,
 ): Promise<string> {
-  const { dappRequest } = request;
-  const { id: actionId, method, origin } = dappRequest;
-  const { securityAlertResponse, traceContext } = dappRequest;
+  const { dappRequest, requestContext } = request;
+  const { id, method } = dappRequest;
+  const actionId = String(id);
+
+  // TODO: Find a home for and define the appropriate MiddlewareContext type
+  const origin = requestContext.assertGet('origin') as string;
```

Comment:
Isn't it just a matter of using the expected type for the context on line 72 above? I am not following why we need casts here either.